lib/sidekiq/job.rb

```diff
@@ -366,9 +366,31 @@ def client_push(item) # :nodoc:
 
       def build_client # :nodoc:
         pool = Thread.current[:sidekiq_redis_pool] || get_sidekiq_options["pool"] || Sidekiq.default_configuration.redis_pool
-        client_class = get_sidekiq_options["client_class"] || Sidekiq::Client
+        client_class = Thread.current[:sidekiq_client_class] || get_sidekiq_options["client_class"] || Sidekiq::Client
         client_class.new(pool: pool)
       end
+
+      # Allows to override the client class used by jobs
+      # enqueued from within the block.
+      #
+      #  MyJob.via_client(CustomClient) do
+      #    MyOtherJob.perform_async(1,2,3)
+      #    SomeJob.perform_bulk([[1], [2], [3]])
+      #    MyJob.perform_async(1,2,3)
+      #  end
+      #
+      # This is useful when you want to use a custom client class
+      # in an ad-hoc manner in a specific context, while still
+      # using the default client class for the rest of the application.
+      def via_client(client_class)
```

Comment:
I don't feel strongly about this convenience method, at least I don't need it for my use-case. Added it primarily for documentation purposes, can remove it if it's unnecessary.

Reply:
I'd rather sharp edges like this not be documented at all. If you really need it, you'll find it.

Reply:
Removed the method and tests [d9426b8](https://github.com/sidekiq/sidekiq/pull/6297/commits/d9426b8ddcd6f3a716cf66bf7b28105fded79997).